content/python/concepts/unittest-module/unittest-module.md

```diff
@@ -0,0 +1,108 @@
+---
+Title: 'Unittest Module'
+Description: 'The unittest module provides features for unit testing Python code.'
+Subjects:
+  - 'Computer Science'
+  - 'Data Science'
+Tags:
+  - 'Automation'
+  - 'Debugging'
+  - 'Development'
+  - 'Modules'
+  - 'Units'
+CatalogContent:
+  - 'learn-python-3'
+  - 'paths/computer-science'
+---
+
+The **`unittest`** [module](https://www.codecademy.com/resources/docs/python/modules) provides features for [unit testing](https://www.codecademy.com/resources/docs/general/unit-testing) Python code. Similar to unit testing frameworks in other languages ,`unittest` allows for the automation of unit tests, sharing of setup and shutdown code for tests, and aggregation of tests into collections.
+
+## Example
+
+Suppose the following code for a `Rectangle` [class](https://www.codecademy.com/resources/docs/python/classes) needs to be tested:
+
+```py
+# saved as rectangle.py
+import math
+
+class Rectangle():
+  def __init__(self, h = 0, w = 0):
+    self.height = h
+    self.width = w
+
+  def area(self):
+    return self.height * self.width
+
+  def perimeter(self):
+    return 2 * (self.height + self.width)
+
+  def isSquare(self):
+    return self.height == self.width
+```
+
+The `unittest` module can set up automated tests for various conditions:
+
+```py
+# saved as test.py
+import unittest
+import rectangle as rect
+
+class RectangleTestCase(unittest.TestCase):
+
+  def test_area(self):
+    self.assertEqual(self.rectangle.area(), 50)
+
+  def test_perimeter(self):
+    self.assertEqual(self.rectangle.perimeter(), 30)
+
+  def test_isSquare(self):
+    self.assertFalse(self.rectangle.isSquare())
+    self.rectangle.width = 5
+    self.assertTrue(self.rectangle.isSquare())
+
+  def test_error(self):
+    self.rectangle.width = "A"
+    with self.assertRaises(TypeError):
+      a = self.rectangle.perimeter()
+
+  def setUp(self):
+    self.rectangle = rect.Rectangle(5 ,10)
+
+if __name__ == '__main__':
+  unittest.main()
+```
+
+First the `unittest.TestCase` class is subclassed for our test case object. Then tests are defined as methods with names starting with `test`. This naming convention allows the test runner to find the methods that represent tests.
```

Comment:
It should be mentioned that `test` is mandatory as the first word in the test function's name.